### PR TITLE
[RFC] Remove "easy" mode.

### PR DESCRIPTION
**Pull request by fmoralesc**
This removes vim's "easy" mode, as discussed in issue #1646:
- [x] remove the `-y` argument ("why" indeed)
- [x] remove support for evim and eview aliases
- [x] remove supporting files (`runtime/evim.vim`)
- [x] remove references to easy mode from the documentation
- [x] update man pages.

### CODE REVIEW

**Comment by Pyrohh** [comment]
It's been just under a month since the last comment, so does anyone think more discussion is needed?

**Comment by justinmk** [comment]
I will review this thread again tonight.

**Comment by fmoralesc** [comment]
I'll rebase ASAP. I'm currently working on the other set of patches at #1793 .

**Comment by Pyrohh** [comment]
Thanks.

**Comment by Pyrohh** [comment]
Is there any reason you're not touching `evim.vim`?

**Comment by fmoralesc** [comment]
Must have slipped through.

**Comment by fmoralesc** [comment]
Any comments on this, then, @justinmk, @Pyrohh? This is ready on my end.

**Comment by Pyrohh** [comment]
Assuming you haven't changed anything recently, then LGTM

**Comment by fmoralesc** [comment]
No, I haven't. I've been pretty busy digging myself out of the pit of #1793. :/

File: src/nvim/main.c [hunk 1]
```diff
@@ -927,9 +926,6 @@ static void parse_command_name(mparm_T *parmp)
   if (parse_char_i(&initstr, 'r'))
```
Comment (justinmk) [inline src/nvim/main.c]: @Pyrohh @fmoralesc I assume line 906 was moved after the rebase, so this comment was addressed?

Comment (Pyrohh) [inline src/nvim/main.c]: Yes, it's about 10 lines before currently.

**Pull request merged**